lib/model/model.go

```diff
@@ -1270,10 +1270,15 @@ func (m *Model) Closed(conn protocol.Connection, err error) {
 	m.pmut.Unlock()
 
 	l.Infof("Connection to %s at %s closed: %v", device, conn.Name(), err)
-	events.Default.Log(events.DeviceDisconnected, map[string]string{
-		"id":    device.String(),
-		"error": err.Error(),
-	})
+
+	data := map[string]string{
```

Comment:
It should be safe to assume that we never close with a nil error - if the error was nil, the connection would not have been closed. I'd even throw in an `if err == nil { panic("bug:...") }` to that effect. Clearly you've *seen things* I haven't, though?

Reply:
Agree. The bug is clearly in the protocol package.